WebShop/WebShop.Frontend/Frontend/src/pages/ItemPage.js

```diff
@@ -1,99 +1,85 @@
+            user: {},
+            quantity: 1
+        }
     }
-    
-    // Item id from route
-    // const params = useParams()
 
-    // Fetch Item by id
-    // const fetchItem = async (id) => {
-    //     const res = await fetch(`http://localhost:5000/tasks/${id}`)
-    //     const data = await res.json()
-    //     return data
-    // }
+    componentDidMount() {
+        axios.get(process.env.REACT_APP_WEB_SHOP_PRDUCTS_BACKEND_API + `/` + this.props.match.params.itemId)
+            .then((res) => { 
+                this.setState({
+                    product : res.data,
+                }); 
+                
+                // Uncomment when Users API are finished
 
-    // Fetch User by userId in Item 
-    // const fetchUser = async (userId) => {
-    //     const res = await fetch(`http://localhost:5000/tasks/${userId}`)
-    //     const data = await res.json()
-    //     return data
-    // }
+                // axios.get(process.env.REACT_APP_WEB_SHOP_USERS_BACKEND_API + `/` + this.state.product.userId)
+                // .then((res2) => { 
+                // this.setState({
+                //     user : res2.data,
+                // }); 
+                // });
+            });
+    }
 
-    const user = {
-        imgSrc: 'https://upload.wikimedia.org/wikipedia/en/thumb/c/c7/Michael_Jordan_crying.jpg/220px-Michael_Jordan_crying.jpg',
-        name: 'Dzordan Dzordanic',
-        email: 'michaeljordan@email.com',
-        phoneNumber: '+381623456789'
+    buy() {
+        console.log(this.state);
+        // Redirect to Paypal
     }
 
-    return (
 
-        <div className="page-div page" style={{backgroundColor:'black'}}>
-            {/* <MainBar/> */}
-            {/* <div className="row center my-2" style={{width:'100%'}}>
-                <div className="col-3" />
-                <Nav className="col" style={{alignItems:'center'}}>
-                    <Nav.Link href="#query=?">New Items</Nav.Link>
-                    <Nav.Link href="#query=?">Best Rated Items</Nav.Link>
-                    <input className="mx-3" type="text" placeholder="Enter query" />
-                    <button type="button" className="btn btn-outline-light" >Search</button>
-                </Nav>
-                <div className="col-3" />
-            </div> */}
+    render() {
+        return (
+            <div className="page-div page" style={{backgroundColor:'black'}}>
             <div className="container my-3">
                 <div className="row">
                     <div className="col"/>
                     <div className="col-10">
                         <div className="row" style={{border:'1px solid white', borderRadius:'10px',  overflow: 'hidden', textOverflow: 'ellipsis'}}>
                             <div className="col-4">
                                 <img 
-                                    src={item.placeHolderImg} 
+                                    src={this.state.product.imageURL} 
```

Comment:
Za sada ok, al sve ove slike treba skinuti sa backenda pa prikazivati ovde. Ako budemo ikad napravili da se stvljaju novi proizvodi i to..

Reply:
Pa i skinuta je sa bekenda. U bazi se cuva taj imageURL. E sad on je tipa string. Video sam implementaciju da slike treba cuvati u bazi kao niz bajtova tako da tu treba dobrano razmisliti kako bi se uglavio taj podatak u atribut src.

Reply:
Trenutno Webshop backend ima hard coded URL slike proizvoda, to će se promeniti u budućnosti.